Add parse-prefer-header definition

XVincentX opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

peterblazejewicz commented on types/parse-prefer-header/index.d.ts:
```diff
@@ -0,0 +1,10 @@
+// Type definitions for node-persist 1.0.0
```
You would have to fix the version to be `major.minor` only to pass the CI test.
You could run same test locally calling:
```bash
npm i
```
and:
```bash
npm run lint parse-prefer-header
```
```bash
npm test
```

peterblazejewicz commented on types/parse-prefer-header/parse-prefer-header-tests.ts:
```diff
@@ -0,0 +1,3 @@
+import parsePreferHeader = require('parse-prefer-header');
+
+parsePreferHeader('Prefer: code=500, value=50');
```
those test could could probably be amended to cover:
```ts
parsePreferHeader(['respond-async, wait=100', 'handling=lenient']);
parse('')
```
as these seems to be supported input types:
https://github.com/ppaskaris/node-parse-prefer-header/blob/master/test.js
Support for `null`/`undefined` input is optional:
```ts
parse(null)
parse(undefined)
```
as probably not intended (but still tested in linked tests). Up to you with those last two.

XVincentX commented:
@peterblazejewicz 

Thanks for the feedback — I started this PR and did not have the time to bring it over the finish line. I've addressed your comments.

sheetalkamat commented on types/parse-prefer-header/index.d.ts:
```diff
@@ -0,0 +1,8 @@
+// Type definitions for node-persist 1.0
```
Name of the package here should be `parse-prefer-header`

XVincentX replied:
Fixed in 3e11647639

XVincentX commented:
//cc @marcelltoth
